### ISSUE TITLE
[Bug] Tile unary ops drop TileView valid_shape and generate mismatched validCol on NPU

### ISSUE THREAD

**Issue by high-cloud** [hw-native-sys/pypto#1370]
### Component

Codegen

### Description

A tile produced from `pl.slice(..., valid_shape=[...])` preserves its `TileView(valid_shape=...)` through `pl.tile.muls`, but the next unary tile op (`pl.tile.neg`) drops the `TileView`. The generated NPU C++ then calls `TNEG(dst, src)` with mismatched valid shape metadata: the source tile has `validCol=4`, while the destination tile has `validCol=8`.

This appears to be a PyPTO/codegen/type propagation issue because the invalid state is already visible in the generated PyPTO IR before PTOAS runs. PTOAS/PTO-ISA could still add a defensive validation for this mismatch, but the root cause is that PyPTO does not preserve `TileView(valid_shape)` across the unary op.

Related but not duplicate: #594 is about automatic valid_shape inference; #1031 is about odd-dimension split with padding + valid_shape.

### Steps to Reproduce

Minimal repro:

```python
import torch
import pypto
import pypto.language as pl
from pypto_lib import compile_and_run


@pypto.jit(
    kernel_kind=pypto.KernelKind.MIX,
    out_shape=[
        [16, 4],  # mix_dbg
        [16, 4],  # plain_raw
        [16, 4],  # valid_raw
        [16, 4],  # valid_scaled
        [16, 4],  # valid_neg
        [16, 4],  # plain_sigmoid
        [16, 4],  # valid_sigmoid
    ],
    out_dtype=[pl.FP32] * 7,
)
def valid_shape_slice_repro(
    x: pl.Tensor[[16, 32], pl.FP32],
    mix_dbg: pl.Out[pl.Tensor[[16, 4], pl.FP32]],
    plain_raw: pl.Out[pl.Tensor[[16, 4], pl.FP32]],
    valid_raw: pl.Out[pl.Tensor[[16, 4], pl.FP32]],
    valid_scaled: pl.Out[pl.Tensor[[16, 4], pl.FP32]],
    valid_neg: pl.Out[pl.Tensor[[16, 4], pl.FP32]],
    plain_sigmoid: pl.Out[pl.Tensor[[16, 4], pl.FP32]],
    valid_sigmoid: pl.Out[pl.Tensor[[16, 4], pl.FP32]],
) -> pl.Tensor[[16, 4], pl.FP32]:
    plain_tile: pl.Tile[[16, 8], pl.FP32] = pl.slice(x, [16, 8], [0, 0])
    valid_tile: pl.Tile[[16, 8], pl.FP32] = pl.slice(
        x, [16, 8], [0, 0], valid_shape=[16, 4]
    )

    plain_calc: pl.Tile[[16, 8], pl.FP32] = pl.sigmoid(plain_tile * 0.5) + 1e-6
    valid_scaled_tile: pl.Tile[[16, 8], pl.FP32] = valid_tile * 0.5
    valid_neg_tile: pl.Tile[[16, 8], pl.FP32] = -valid_scaled_tile
    valid_calc: pl.Tile[[16, 8], pl.FP32] = pl.sigmoid(valid_scaled_tile) + 1e-6

    mix_dbg = pl.store(valid_scaled_tile, [0, 0], mix_dbg)
    plain_raw = pl.store(plain_tile, [0, 0], plain_raw)
    valid_raw = pl.store(valid_tile, [0, 0], valid_raw)
    valid_scaled = pl.store(valid_scaled_tile, [0, 0], valid_scaled)
    valid_neg = pl.store(valid_neg_tile, [0, 0], valid_neg)
    plain_sigmoid = pl.store(plain_calc, [0, 0], plain_sigmoid)
    valid_sigmoid = pl.store(valid_calc, [0, 0], valid_sigmoid)
    return mix_dbg


def golden(x: torch.Tensor):
    x_valid = x[:, :4]
    valid_scaled = x_valid * 0.5
    return (
        valid_scaled,
        x_valid,
        x_valid,
        valid_scaled,
        -valid_scaled,
        torch.sigmoid(valid_scaled) + 1e-6,
        torch.sigmoid(valid_scaled) + 1e-6,
    )


if __name__ == "__main__":
    x = torch.arange(16 * 32, dtype=torch.float32).reshape(16, 32) / 64 - 2
    compile_and_run(valid_shape_slice_repro, [x], golden(x), rtol=1e-5, atol=1e-5)
```

Run on sim:

```bash
python examples/debug/valid_shape_slice_repro.py -p a2a3sim
```

Run on NPU:

```bash
task-submit --device auto --env PYTHONPATH --env ASCEND_HOME_PATH --env PTOAS_ROOT --env PTO_ISA_ROOT --run "python examples/debug/valid_shape_slice_repro.py -p a2a3 -d {}"
```

### Expected Behavior

The valid-shape tile path should match the full tile path for the first 4 columns:

- `valid_raw == x[:, :4]`
- `valid_scaled == x[:, :4] * 0.5`
- `valid_neg == -(x[:, :4] * 0.5)`
- `valid_sigmoid == sigmoid(x[:, :4] * 0.5) + 1e-6`

Unary/same-shape elementwise tile ops should preserve the input `TileView(valid_shape=[16, 4])` on their output tile, or the lowered destination tile should use the same runtime valid shape as the source.

### Actual Behavior

On `a2a3sim`, all outputs pass.

On real NPU, `valid_raw` and `valid_scaled` pass, but `valid_neg` is the first failing output and `valid_sigmoid` also fails:

```text
[RUN]   'mix_dbg' PASS  shape=(16, 4) dtype=torch.float32
[RUN]   'plain_raw' PASS  shape=(16, 4) dtype=torch.float32
[RUN]   'valid_raw' PASS  shape=(16, 4) dtype=torch.float32
[RUN]   'valid_scaled' PASS  shape=(16, 4) dtype=torch.float32
[RUN]   'valid_neg' FAIL  shape=(16, 4) dtype=torch.float32
[RUN]   'plain_sigmoid' PASS  shape=(16, 4) dtype=torch.float32
[RUN]   'valid_sigmoid' FAIL  shape=(16, 4) dtype=torch.float32
Output(s) does not match golden: ['valid_neg', 'valid_sigmoid']
  'valid_neg' FAIL  shape=(16, 4) dtype=torch.float32
    Mismatched elements: 64/64  rtol=1e-05 atol=1e-05
    first 20 mismatches:
    [0] actual=0.2689414322376251, expected=1.0
    [1] actual=0.270480215549469, expected=0.9921875
    [2] actual=0.27202457189559937, expected=0.984375
    [3] actual=0.2735744118690491, expected=0.9765625
```

Generated IR evidence from `passes_dump/*_after_Simplify.py`:

```python
valid_tile__tile: pl.Tile[[16, 8], pl.FP32, ..., pl.TileView(valid_shape=[16, 4])] = pl.tile.load(...)
valid_scaled_tile__tile: pl.Tile[[16, 8], pl.FP32, ..., pl.TileView(valid_shape=[16, 4])] = pl.tile.muls(valid_tile__tile, 0.5)
valid_neg_tile__tile: pl.Tile[[16, 8], pl.FP32, ..., pl.Mem.Vec] = pl.tile.neg(valid_scaled_tile__tile)
```

Generated C++ evidence:

```cpp
Tile<..., 16, 8, ..., -1, -1, ...> v46(...)(v16, v12);  // source validCol = 4
TMULS(v46, v29, v11);

Tile<..., 16, 8, ..., -1, -1, ...> v48(...)(v16, v13);  // destination validCol = 8
TNEG(v48, v46);
```

So `TNEG` receives a source tile with `validCol=4` and a destination tile with `validCol=8`. This mismatch is not rejected in the NPU run and produces incorrect output.

### Git Commit ID

`dfb7321d87b19c068852ca3b0a05c3c270b924f6`

### NPU Kind

Ascend 910B

### Host Platform

Linux (aarch64)

### Additional Context

Environment used to reproduce:

- pypto: `dfb7321d87b19c068852ca3b0a05c3c270b924f6` on `main`
- simpler/runtime: `a94d51407ba13206439dfda25266c1d10d78be5a`
- pypto-lib repro host commit: `13990a5`
- CANN: `8.5.0.alpha001`
- ptoas: `0.38`
- Python: `3.10.20`
- Task id for the failing NPU repro: `task_20260514_171354_26356316509`

This was originally found while optimizing `models/deepseek/v4/hc_pre.py` in pypto-lib. Removing `valid_shape` from the problematic slice is a local workaround, but it loses the intended valid-shape semantics and should not be required.
